src/Component/Component.php

```diff
@@ -16,6 +16,53 @@
 
 namespace Zest\Component;
 
+use Zest\http\Request;
+use Zest\http\Response;
+use Zest\Input\Input;
+
 class Component extends \Zest\Router\Router
 {
+    /**
+     * Dispatch the route, creating the controller object and running the
+     * action method.
+     *
+     * @param string $url The route URL
+     *
+     * @since 3.0.0
+     *
+     * @return void
+     */
+    public function dispatch(Request $request)
+    {
+        $url = $request->getQueryString();
+        $url = $this->RemoveQueryString($url, new Request());
+        if ($this->match($url)) {
+            (isset($this->params['middleware'])) ? $this->params['middleware'] = new $this->params['middleware']() : null;
+            if (!isset($this->params['callable'])) {
+                $controller = $this->params['controller'];
+                $controller = $this->convertToStudlyCaps($controller);
+                $controller = $this->getNamespace().$controller;
+                if (class_exists($controller)) {
+                    (isset($this->params['middleware']) && is_object($this->params['middleware'])) ? ( new $this->params['middleware']())->before(new Request(), new Response(), $this->params) : null;
+                    $controller_object = new $controller($this->params, $this->getInput(new Input()));
+                    $action = $this->params['action'];
+                    $action = $this->convertToCamelCase($action);
+                    if (preg_match('/action$/i', $action) == 0) {
+                        $controller_object->$action();
+                        (isset($this->params['middleware']) && is_object($this->params['middleware'])) ? (new $this->params['middleware']())->after(new Request(), new Response(), $this->params) : null;
+                    } else {
+                        throw new \Exception("Method $action in controller $controller cannot be called directly - remove the Action suffix to call this method");
+                    }
+                } else {
+                    throw new \Exception("Controller class $controller not found");
```

Comment:
If the route is not matched, it should send the `404` HTTP status code.

It means the request URL is not found.

Reply:
I don't think that throwing exception directly is a good way to solve this issue.

Reply:
yes, but what about if user want to customize this like in Laravel
http://laravel.com/dadasd

Reply:
oh @peter279k sorry for misunderstand we can fetch Views from View folder

Reply:
Just return view with customized error page :).

Reply:
Perhaps we can pass the default error page with view.

And let users customize their error pages.

Reply:
@peter279k  thanks